tests/e2e/statefulset_with_topology.go

```diff
@@ -129,4 +129,82 @@ var _ = ginkgo.Describe("[csi-topology-vanilla] Topology-Aware-Provisioning-With
 		fss.DeleteAllStatefulSets(client, namespace)
 	})
 
+	/*
+		Test to verify PV node affinity details such that it should contain both zone and
+		region details of SC.
+		Verify POD is running on the same node as mentioned in node affinity details.
+		Steps
+		1. Create SC with zone and region details specified in the SC
+		2. Create statefulset with replica 3 with the above SC
+		3. Wait for all the stateful set to be up and running
+		4. Wait for pvc to be in bound state and statefulset pod in running state
+		5. Describe pv and verify node affinity details should contain both zone and region details
+		6. Verify POD is running on the same node as mentioned in node affinity details
+		7. Delete statefulset
+		8. Delete PVC and SC
+	*/
+
+	ginkgo.It("Verify pod and pv topology affinity details", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		// Creating StorageClass
+		ginkgo.By("Creating StorageClass for Statefulset")
+		scSpec := getVSphereStorageClassSpec(defaultNginxStorageClassName, nil, allowedTopologies, "", "", false)
+		sc, err := client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		// Creating statefulset with 3 replicas
+		ginkgo.By("Creating statefulset with 3 replica")
+		statefulset := GetStatefulSetFromManifest(namespace)
+		ginkgo.By("Creating statefulset")
+		CreateStatefulSet(namespace, statefulset, client)
+		replicas := *(statefulset.Spec.Replicas)
+
+		// Waiting for pods status to be Ready.
+		fss.WaitForStatusReadyReplicas(client, statefulset, replicas)
+		gomega.Expect(fss.CheckMount(client, statefulset, mountPath)).NotTo(gomega.HaveOccurred())
+		ssPodsBeforeScaleDown := fss.GetPodList(client, statefulset)
+		gomega.Expect(ssPodsBeforeScaleDown.Items).NotTo(gomega.BeEmpty(),
+			fmt.Sprintf("Unable to get list of Pods from the Statefulset: %v", statefulset.Name))
+		gomega.Expect(len(ssPodsBeforeScaleDown.Items) == int(replicas)).To(gomega.BeTrue(),
+			"Number of Pods in the statefulset should match with number of replicas")
+
+		// Verify node and pv topology affinity should contains specified zone and region details of SC
```

Comment:
Going forward ,  validation of node affinity labels on PV and checking POD running on the expected node is common validation steps.  Instead of writing it in every testcase , can we move it to a generic method. 

Make sure we pass statefulset and expected zone/region as an argument to the method

Reply:
acknowledged.